jina/serve/runtimes/worker/__init__.py

```diff
@@ -194,6 +194,8 @@ async def async_cancel(self):
         self.logger.debug('cancel WorkerRuntime')
         if self._hot_reload_task is not None:
             self._hot_reload_task.cancel()
+        await self._request_handler.close()  # allow pending requests to be processed
```

Comment:
no, u should enter the health servicer into graceful shutdown before closing the request so that u announce u are shutting down to the outside world

Reply:
this might mean orchestrators kill the running pod right ?

Reply:
what do u mean?

Reply:
yes, it may mean this